src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -556,12 +621,27 @@ void DlgPrefSound::masterLatencyChanged(double latency) {
     update();
 }
 
-void DlgPrefSound::headDelayChanged(double value) {
+void DlgPrefSound::masterDelaySpinboxChanged(double value) {
+    m_pMasterDelay->set(value);
+}
+
+void DlgPrefSound::headDelaySpinboxChanged(double value) {
     m_pHeadDelay->set(value);
 }
 
-void DlgPrefSound::masterDelayChanged(double value) {
-    m_pMasterDelay->set(value);
+void DlgPrefSound::boothDelaySpinboxChanged(double value) {
+    m_pBoothDelay->set(value);
+}
+
+void DlgPrefSound::roundTripLatencySpinboxChanged(double value) {
+    m_pRoundTripLatency->set(value);
+    roundTripLatencyChanged(value);
+}
+
+void DlgPrefSound::roundTripLatencyChanged(double newRoundTripLatency) {
+    double inputLatencyCompensation = newRoundTripLatency / 2.0;
```

Comment:
It does not matter when the signal is mixed in internally. It does matter that the user speaks normally in sync with the sound he hears on the headphone. 

If the user flicks in sync with a beat on the headphone, this flick can be heard after a complete roundtrip in the master output (isolated from the flicking user) so the content heard on the headphone must be passed also with the same complete roundtrip delay to the master. 

Actually the ouput delay of the master does not matter at all, because it effects the beat and the flick in the same way. What counts is the time the flick reaches the engine plus the time the beat needs from the engine to the users ears. If one of these times is changed the delay compensation has to be changed as well.